souffle-lang/souffle
# Update Travis scripts

Update Travis scripts
- add Stages (style, testing, and packaging stages)
- use Travis container builds for faster job startup (which can be improved further when Travis finally adds 16.04)
- remove unnecessary test options
- add osx gcc
- split multi-purpose scripts into single purpose scripts
- use gcc7 and clang4, the current commonly installed versions
- bugfixes for problems revealed in the update
- update style to match clang-format-4.0 (fixing a bugged style in 3.8)
- add check for the required bison 3.0.2

Note that bar a bugfix in mcpp, every src file changed is a style fix.
